File: Second Year (2013-2014)/OMGL7/TP3 - Website in JSP/src/db/CategorieDAO.java
package db;

import java.sql.*;
import java.util.ArrayList;
/**
* @author Andriamilanto Tompoariniaina - Canno Camille - Info2B
*/

/**
* Classe pour les objets DAO de Categorie
*/
public class CategorieDAO {
//Methodes
	/**
	*Constructeur
	*/
	public CategorieDAO() {}

	/**
	*Pour retrouver tous les tuples
	*/
	public ArrayList<Categorie> findAll() {
		Statement stat = null;
		String query = "";
		ArrayList<Categorie> ret = new ArrayList<Categorie>();

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Categorie;";

			//Le resultat a retourner
			ResultSet res = stat.executeQuery(query);

			//Mettre le resultat de chacun des tuples de la requete sous la forme d'un Array
			while (res.next()) {
				Categorie cat = new Categorie();
				cat.init(res.getInt(1), res.getString(2));
				ret.add(cat);
			}
		}
		catch (SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		//Retourne l'execution de la requete sous la forme d'un objet ResultSet
		return ret;
	}

	/**
	*Permet de retrouver juste un tuple
	*@param id idCategorie de la categorie a retrouver
	*/
	public Categorie find(int id) {
		Statement stat = null;
		String query = "";
		Categorie ret = new Categorie();

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Categorie WHERE IdCategorie = " + id + ";";

			//Retourne l'execution de la requete sous la forme d'un objet ResultSet
			ResultSet result = stat.executeQuery(query);

			//Si le resultat est bon, prends la premiere ligne
			if (result.first()) {
				ret.init(id, result.getString(2));
			}
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		return ret;
	}


	/**
    *Methode qui permet d'inserer un tuple
    *@param tuple Objet de type Categorie a inserer
    */
    public void insert(Categorie tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Categorie
        int idC = tuple.getIdCategorie();
        String lib = tuple.getLibelleCategorie();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "INSERT INTO Categorie VALUES(" + idC + "," + lib + ");";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de faire la mise a jour d'un tuple
    *@param tuple Objet nouveau a mettre a la place de l'ancien
    */
    public void update(Categorie tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Categorie
        int idC = tuple.getIdCategorie();
        String lib = tuple.getLibelleCategorie();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "UPDATE Categorie SET LibelleCategorie = " + lib + " WHERE IdCategorie = " + idC + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de supprimer un tuple
    *@param idCategorie Id du tuple a supprimer
    */
    public void delete(int idCategorie) {
    	Statement stat = null;
        String query = "";

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "DELETE FROM Categorie WHERE IdCategorie = " + idCategorie + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }
}

File: Second Year (2013-2014)/OMGL7/TP2 - Website in PHP/musiconline/musiconline/db/CategorieDao.php
<?php
/* Authors: ANDRIAMILANTO Tompoariniaina && CANNO Camille
* Groupe: B1
* File: CategorieDao.php
*/

import ('phpens.framework.Application');
import ('phpens.framework.db.DataObject');
import ('phpens.framework.db.DataAccessObject');
import ('musiconline.db.Categorie');


class CategorieDao implements DataAccessObject {
  //Attribut instance de DataAccessObject
  private static $dao;

  //Constructeur vide
  private function __construct() { }

  //Retourne l'instance de CategorieDAO
  public final static function getInstance() {

    //Vérifie si l'instance est correcte ou non
    if(!isset(self::$dao)) {

      //S'il n'en est pas, il en créer un autre sans paramètres
      self::$dao= new CategorieDao();
    }

    //S'il en est un, il le retourne, sinon il retourne donc une instance vide
    return self::$dao;
  }


  //Retourne toute la table
  public final function findAll() {

    //Récupère la méthode de connection à la Base de donnée
    $dbc = Application::getInstance()->getDBConnection();

    //Variable où la requête est enregistrée
    $query = "select * from Categorie";

    //Requête en dur, prends la bdd et y exécute la requête
    $stmt = $dbc->query($query);

    /*Récupère tout les tuples de Categorie, et les transpose dans des instances de classe Categorie
     Et fini par enregistrer le résultat dans $results, FALSE est retourné s'ya erreur */
    $results = $stmt->fetchALL(PDO::FETCH_CLASS, 'Categorie');

    //Retourne le résultat de $results
    return $results;
   }


  //Permet d'ajouter un tuple
  public final function insert(DataObject $do) {

    //Vérifie si l'objet entré en paramètre est instance de Categorie
    if($do instanceof Categorie){

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "insert into Categorie(IdCategorie,LibelleCategorie) values (:IdCategorie,:LibelleCategorie)";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':IdCategorie',$do->getIDCategorie(),PDO::PARAM_STR);
      $stmt->bindValue(':LibelleCategorie',$do->getLIbelleCategorie(),PDO::PARAM_STR);

      //Exécute finallement la requête
      $stmt->execute($data);
   }
 }

  //Permet de mettre à jour un tuple
  public function update(DataObject $obj) {

    //Vérifie si l'objet entré en paramètre est instance de Album
    if($obj instanceof Categorie){

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "UPDATE Categorie SET LibelleCategorie = :LibelleCategorie WHERE IdCategorie = :IdCategorie";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':IdCategorie',$obj->getIDCategorie(),PDO::PARAM_STR);
      $stmt->bindValue(':LibelleCategorie',$obj->getLIbelleCategorie(),PDO::PARAM_STR);

      //Exécute finallement la requête
      $stmt->execute($data);
   }
  }


  //Permet de supprimer un tuple
  public function delete(DataObject $obj) {

    //Vérifie si l'objet entré en paramètre est instance de Album
	  if($obj instanceof Categorie){

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "DELETE FROM Categorie WHERE IdCategorie = :IdCategorie";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':IdCategorie',$obj->getIDCategorie(),PDO::PARAM_STR);

      //Exécute finallement la requête
      $stmt->execute($data);
    }
  }

}
?>
